pallets/usage/src/lib.rs

```diff
@@ -0,0 +1,384 @@
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Gas and value transfer currency
+        type Currency: Currency<Self::AccountId> + ReservableCurrency<Self::AccountId>;
+
+        /// Gas to Currency converter
+        type GasConverter: GasToFeeConverter<Balance = BalanceOf<Self>>;
+
+        /// Type providing interface to make payment in currency units
+        type PaymentProvider: PaymentProvider<Self::AccountId, Balance = BalanceOf<Self>>;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+
+        /// The desired interval between offchain worker invocations.
+        #[pallet::constant]
+        type WaitListTraversalInterval: Get<Self::BlockNumber>;
+
+        /// Time lock expiration duration for an offchain worker
+        #[pallet::constant]
+        type ExpirationDuration: Get<u64>;
+
+        /// The desired interval between offchain worker invocations.
+        #[pallet::constant]
+        type MaxBatchSize: Get<u32>;
+
+        /// The amount of gas necessary for a trap reply message to be processed.
+        #[pallet::constant]
+        type TrapReplyExistentialGasLimit: Get<u64>;
+
+        /// The fraction of the collected wait list rent an external submitter will get as a reward
+        #[pallet::constant]
+        type ExternalSubmitterRewardFraction: Get<Perbill>;
+    }
+
+    type BalanceOf<T> =
+        <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        WaitListRentCollected,
+    }
+
+    // Gear pallet error.
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Value not found for a key in storage.
+        FailedToGetValueFromStorage,
+    }
+
+    /// Methods for the `ValidateUnsigned` implementation:
+    /// Restricts calls to `collect_waitlist_rent_unsigned` to local calls
+    /// (i.e. extrinsics generated on this node) or those already in a block
+    /// therefore ruling out any origin other than block authors.
+    #[pallet::validate_unsigned]
+    impl<T: Config> ValidateUnsigned for Pallet<T> {
+        type Call = Call<T>;
+        fn validate_unsigned(source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+            match call {
+                Call::collect_waitlist_rent { payees_list: _ } => {
+                    // Only accept transactions from a trusted source: local OCW or already in block.
+                    match source {
+                        TransactionSource::Local | TransactionSource::InBlock => { /* allowed */ }
+                        _ => return InvalidTransaction::Call.into(),
+                    }
```

Comment:
Maybe to use `matches!`?
```suggestion
                    if !matches!(source, TransactionSource::Local | TransactionSource::InBlock) {
                        return InvalidTransaction::Call.into(),
                    }
```